Add @PartFilename config - on/off, value, value suffix

patr1kt0th opened the pull request:
Fixes: https://github.com/quarkiverse/quarkus-openapi-generator/issues/435

hbelmiro commented:
@patr1kt0th for you to run the RESTEasy Reactive tests locally:
```shell
cd integration-tests
mvn clean install -Presteasy-reactive
mvn verify -Presteasy-reactive
```

patr1kt0th commented:
@hbelmiro thanks, I've already prepared a "fix" and split the test classes (classic & reactive), but I was/am fighting with a compilation error, so I'll try your suggestion this evening and let's see... will get back to you as soon as I have something new

patr1kt0th commented:
@hbelmiro, @ricardozanini, please check. It worked locally, so hopefully there won't be any problems.

But I have one question... I did the new config options in a similar way the others are done, but I was thinking whether it wouldn't be better to put all _PartFilename_ related config under a `part-filename` key and use/rename them for example like this:

- `quarkus.openapi-generator.codegen.(spec.xyz).partfile-name.generate=true|false`
- `quarkus.openapi-generator.codegen.(spec.xyz).partfile-name.value=<some-value>`
- `quarkus.openapi-generator.codegen.(spec.xyz).partfile-name.add-field-name-to-value=true|false` (because I don't like the `part-filename-value-suffix` that much)

What do you think?

patr1kt0th commented on deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/wrapper/OpenApiClientGeneratorWrapper.java:
```diff
@@ -179,6 +179,21 @@ public void withTemplateDir(Path templateDir) {
+        return this;
+    }
+
+    public OpenApiClientGeneratorWrapper withPartFilenameValueSuffixConfig(final String partFilenameValueSiffix) {
```
Typo, it should be `partFilenameValueSuffix` not `partFilenameValueSiffix`

hbelmiro commented on integration-tests/part-filename/pom.xml:
```diff
@@ -0,0 +1,93 @@
+      <groupId>io.quarkiverse.openapi.generator</groupId>
+      <artifactId>quarkus-openapi-generator</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.jboss.resteasy</groupId>
+      <artifactId>resteasy-multipart-provider</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-resteasy-reactive-common</artifactId>
+    </dependency>
```
We declare dependencies that are specific for RESTEasy Reactive or Classic in its equivalent profile in [integration-tests/pom.xml](https://github.com/quarkiverse/quarkus-openapi-generator/blob/5f10fa319d4b58d37ce68e6aaa953796241c11a3/integration-tests/pom.xml#L47). I'm not sure about `resteasy-multipart-provider`, but you can move at least `quarkus-resteasy-reactive-common` to there.

patr1kt0th replied:
Thanks for the info, I noticed it, but I did not want to put them there, as they're (only) relevant to the `part-filename` tests and I didn't want to influence the rest of the tests.

I need them to be able to compile the tests as they're checking both types of annotations and I couldn't figure out, how to do it in a more elegant way. Or maybe I'm just doing something wrong?

Btw. I think the `quarkus-resteasy-reactive-common` should already be part of `quarkus-rest-client-reactive-jackson` which you already have there.

hbelmiro reviewed (commented):
Thank you for this contribution @patr1kt0th. Very good job!
I just left a comment regarding where you placed the dependencies for the test.

And one picky suggestion: it seems you can avoid code duplication in tests. Maybe using parameterized tests or just extracting common parts to a different method. IDK, it's up to you and totally optional. If you want to leave it as it is, that's fine for me.

patr1kt0th commented:
Thanks for the feedback, @hbelmiro. I fully agree with the picky suggestion... I didn't want to spend too much time with it, I assume. But I improved it a bit, please check.

I also fixed the typo and the README's format and I tried to implement the suggestion I made earlier = introducing the `part-filename` config group, but it did not work as I imagined it (it worked, but there were some warnings), so I just renamed the "suffix" config parameter.

patr1kt0th commented:
Oh, I just noticed I messed up the README.md file. @hbelmiro, can you please tell me what kind of MD settings you use in IntelliJ? Is it documented somewhere? Thanks.

hbelmiro commented:
@patr1kt0th we don't have any specific MD settings. As long as you don't mess it up too much, that's fine (which doesn't seem to be the case 😆).
I'll review it ASAP.

patr1kt0th commented:
Thanks @hbelmiro, I reverted the messed-up formatting and hopefully fixed all the issues it caused.

patr1kt0th commented:
Thanks for the review and the suggestions/fixes @ricardozanini 👍

patr1kt0th commented:
Thanks for your cooperation, guys 🤝

hbelmiro commented:
@all-contributors please add @patr1kt0th for code
